added evaluation for 2.5.2 client-constraints (#278)
solid/authorization-panel: proposals/evaluation/uc-2.5.2-client-constraints.md

# Limiting application access while not acting as resource controller

https://solid.github.io/authorization-panel/authorization-ucr/#uc-client-constraints

### ACP

ACP requires that if `acp:agent` and `acp:client` are present in the matcher, both need to be satisfied.

```ttl
PREFIX acp: <http://www.w3.org/ns/solid/acp#>
PREFIX acl: <http://www.w3.org/ns/auth/acl#>
PREFIX ex: <https://example.com/>

ex:ReaderAgentPolicy
  acp:anyOf ex:BobMatcher, ex:AliceMatcher ;
  acp:allow acl:Read .

ex:BobMatcher
  acp:agent ex:Bob ;
  acp:client ex:Projectron .

ex:AliceMatcher
  acp:agent ex:Alice ;
  acp:client ex:Projectron, ex:OtherApp .

> 

Author: ```suggestion
```

```

# Limiting application access while not acting as resource controller

https://solid.github.io/authorization-panel/authorization-ucr/#uc-client-constraints

### ACP

ACP requires that if `acp:agent` and `acp:client` are present in the matcher, both need to be satisfied.

```ttl
PREFIX acp: <http://www.w3.org/ns/solid/acp#>
PREFIX acl: <http://www.w3.org/ns/auth/acl#>
PREFIX ex: <https://example.com/>

ex:ReaderAgentPolicy
  acp:anyOf ex:BobMatcher, ex:AliceMatcher ;
  acp:allow acl:Read .

ex:BobMatcher
  acp:agent ex:Bob ;
  acp:client ex:Projectron .

ex:AliceMatcher
  acp:agent ex:Alice ;

>   acp:client ex:Projectron, ex:OtherApp .
>   

Reviewer: ```suggestion
  acp:client ex:Projectron, ex:OtherApp .
```

# Limiting application access while not acting as resource controller

https://solid.github.io/authorization-panel/authorization-ucr/#uc-client-constraints

### ACP

> ACP allows combining multiple matchers with `acp:allOf`, this allows expressin policy which uses `acp:agent` and `acp:client` in combined matchers.

Reviewer 2: ```suggestion
ACP allows combining multiple matchers with `acp:allOf`, this allows expressing policy which uses `acp:agent` and `acp:client` in combined matchers.
```

> ACP allows combining multiple matchers with `acp:allOf`, this allows expressing policy which uses `acp:agent` and `acp:client` in combined matchers.

Author: ```suggestion
ACP requires that if `acp:agent` and `acp:client` are present in the matcher, both need to be satisfied.
```

```ttl
PREFIX acp: <http://www.w3.org/ns/solid/acp#>
PREFIX acl: <http://www.w3.org/ns/auth/acl#>
PREFIX ex: <https://example.com/>

ex:accessControlResourceA
  acp:resource ex:resourceX ;
  acp:accessControl ex:accessControlB .

ex:accessControlB acp:apply ex:PolicyC .

ex:PolicyC
  acp:allOf ex:agentMatcherD, ex:clientMatcherD ;
  acp:allow acl:Read .

ex:agentMatcherD
  acp:agent ex:Bob .

ex:clientMatcherD

>   acp:client ex:Projectron .

Reviewer 2: Note that this is equivalent effectively to:

```ttl
ex:PolicyC
  acp:allOf ex:matcherD ;
  acp:allow acl:Read .

ex:matcherD
  acp:agent ex:Bob ;
  acp:client ex:Projectron .
```

Which is interesting because we could imagine a future where a matcher dedicated to agent Bob would be maintained by them and they might edit the clients they trust in and out.

For example something such as:

```ttl
ex:ReaderAgentPolicy
  acp:anyOf ex:bobMatcher, ex:aliceMatcher ;
  acp:allow acl:Read .

ex:bobMatcher
  acp:agent ex:Bob ;
  acp:client ex:Projectron .

ex:aliceMatcher
  acp:agent ex:Alice ;
  acp:client ex:Projectron, ex:OtherApp .
```

Obviously, that would be if you trust Alice and Bob to delegate read access. But you could also lock in the fact that Bob has read access and let him handle the clients he trusts via a clientMatcher (and have 1 policy per agent, ex:BobReaderAgentPolicy)...

> ```ttl
> PREFIX acp: <http://www.w3.org/ns/solid/acp#>
> PREFIX acl: <http://www.w3.org/ns/auth/acl#>
> PREFIX ex: <https://example.com/>
> 
> ex:accessControlResourceA
>   acp:resource ex:resourceX ;
>   acp:accessControl ex:accessControlB .
> 
> ex:accessControlB acp:apply ex:PolicyC .
> 
> ex:PolicyC
>   acp:allOf ex:agentMatcherD, ex:clientMatcherD ;
>   acp:allow acl:Read .
> 
> ex:agentMatcherD
>   acp:agent ex:Bob .
> 
> ex:clientMatcherD
>   acp:client ex:Projectron .
> ```

Author: ```suggestion
```ttl
PREFIX acp: <http://www.w3.org/ns/solid/acp#>
PREFIX acl: <http://www.w3.org/ns/auth/acl#>
PREFIX ex: <https://example.com/>

ex:ReaderAgentPolicy
  acp:anyOf ex:BobMatcher, ex:AliceMatcher ;
  acp:allow acl:Read .

ex:BobMatcher
  acp:agent ex:Bob ;
  acp:client ex:Projectron .

ex:AliceMatcher
  acp:agent ex:Alice ;
  acp:client ex:Projectron, ex:OtherApp .
```

> `ex:Policy` allows acces to `ex:Projectron` but only if `ex:Bob` is using it.

Author: ```suggestion
`ex:BobMatcher` allows access to `ex:Projectron` but only if `ex:Bob` is using it.
```
